lib/routes/admin-api/tag.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const Controller = require('../controller');
+
+const { tagSchema } = require('./tag-schema');
+const { CREATE_TAG, DELETE_TAG } = require('../../command-type');
+const { UPDATE_FEATURE } = require('../../permissions');
+const { handleErrors } = require('./util');
+const extractUsername = require('../../extract-user');
+
+const version = 1;
+
+class TagController extends Controller {
+    constructor(config) {
+        super(config);
+        this.featureTagStore = config.stores.featureTagStore;
+        this.eventStore = config.stores.eventStore;
+        this.logger = config.getLogger('/admin-api/tag.js');
+
+        this.delete('/id/:tagId', this.deleteTag, UPDATE_FEATURE);
+        this.get('/id/:tagId', this.getTag);
+        this.get('/', this.getTags);
+        this.post('/', this.createTag, UPDATE_FEATURE);
+        this.get('/:type', this.getTagsByType);
+        this.get('/:type/:value', this.getTagByTypeAndValue);
+    }
+
+    async getTags(req, res) {
+        const tags = await this.featureTagStore.getTags();
+        res.json({ version, tags });
+    }
+
+    async getTagsByType(req, res) {
+        const tags = await this.featureTagStore.getAllOfType(req.params.type);
+        res.json({ version, tags });
+    }
+
+    async getTag(req, res) {
+        try {
+            const id = Number.parseInt(req.params.tagId, 10);
```

Comment:
A dirty shorthand for this is `+req.params.tagId`.

Reply:
Seeing as how you call it dirty, what would you prefer? the shorthand or the explicit? :)